src/main/java/de/zalando/zmon/dataservice/Application.java

```diff
@@ -13,6 +23,28 @@
 @EnableConfigurationProperties
 public class Application {
 
+    private static final String JAEGER = "jaeger";
+    private static final String LIGHTSTEP = "lightstep";
+    private static final String INSTANA = "instana";
+
+    @Autowired
+    private OpenTracingConfigProperties openTracingConfig;
+
+    private final Logger logger = LoggerFactory.getLogger(Application.class);
+
+    @Bean
+    public Tracer tracer() {
+        Tracer tracer = NoopTracerFactory.create();
+        if (JAEGER.equalsIgnoreCase(openTracingConfig.getTracingProvider())){
+            tracer = new JaegerConfig(openTracingConfig).generateTracer();
+        } else if (LIGHTSTEP.equalsIgnoreCase(openTracingConfig.getTracingProvider())){
+                tracer = new LightStepConfig(openTracingConfig).generateTracer();
+        } else if (INSTANA.equalsIgnoreCase(openTracingConfig.getTracingProvider())) {
+            tracer = InstanaTracerFactory.create();
+        }
+        return tracer;
+    }
```

Comment:
while creating the tracer, I initialize it as the noop tracer 

Tracer tracer = NoopTracerFactory.create();

If none of the tracing providers match the if condition then it will be the noop tracer by default